metrics/metrics.go

```diff
@@ -0,0 +1,66 @@
+package metrics
+
+import (
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+)
+
+var (
+	MOutgoingMsgs = stats.Int64("pubsub/outgoing_msg_size", "Outgoing message size", "By")
+	MIncomingMsgs = stats.Int64("pubsub/incoming_msg_size", "Incoming message size", "By")
+	MTopics       = stats.Int64("pubsub/topics", "Number of topics currently subscribed", "1")
+	MPeers        = stats.Int64("pubsub/peers", "Number of pubsub peers", "1")
+
+	OutgoingMsgCountView = &view.View{
+		Name:        "pubsub/outgoing_msg_count",
+		Description: "Number of outgoing messages sent",
+		Measure:     MOutgoingMsgs,
+		Aggregation: view.Count(),
+	}
+
+	OutgoingMsgSizeView = &view.View{
+		Name:        "pubsub/outgoing_msg_size",
+		Description: "Sizes of outgoing messages sent",
+		Measure:     MOutgoingMsgs,
+		Aggregation: view.Distribution(0, 128, 512, 1024, 1024*1024),
+	}
+
+	IncomingMsgCountView = &view.View{
+		Name:        "pubsub/incoming_msg_count",
+		Description: "Number of outgoing messages sent",
+		Measure:     MIncomingMsgs,
+		Aggregation: view.Count(),
+	}
+
+	IncomingMsgSizeView = &view.View{
+		Name:        "pubsub/incoming_msg_size",
+		Description: "Sizes of outgoing messages sent",
+		Measure:     MIncomingMsgs,
+		Aggregation: view.Distribution(0, 128, 512, 1024, 1024*1024),
+	}
+
+	TopicsGaugeView = &view.View{
+		Name:        "pubsub/topics",
+		Description: "Topics the pubsub instance is routing",
+		Measure:     MTopics,
+		Aggregation: view.LastValue(),
+	}
+
+	PeersGaugeView = &view.View{
+		Name:        "pubsub/peers",
+		Description: "Pubsub peers the host is connected to",
+		Measure:     MPeers,
+		Aggregation: view.LastValue(),
+	}
+)
+
+func Register() error {
+	return view.Register(
```

Comment:
Libraries shouldn't register views. That is for the application to do.

Reply:
so the thought is our libraries can expose functions like this to be called from the application, registering all the views in the module. i don't think it's unreasonable, but i'm definitely open to an alternative. this should never be called within the package

Reply:
Tying this in with the upcoming host/service refactor, it could make sense to have a `Measurable` (or like) interface that the omnihost can cast services to and call to register default views, iff a configuration option is enabled (and it’ll probably be enabled by default).

Reply:
An example from the OC GRPC plugin: https://github.com/census-instrumentation/opencensus-go/blob/master/plugin/ocgrpc/server_metrics.go#L91-L97